src/OpenTelemetry/Trace/Builder/TracerProviderBuilderExtensions.cs

```diff
@@ -0,0 +1,355 @@
+            where T : BaseExporter<Activity>
+        {
+            if (tracerProviderBuilder is TracerProviderBuilderBase tracerProviderBuilderBase)
+            {
+                tracerProviderBuilderBase.AddExporter<T>(exportProcessorType, configure);
+            }
+
+            return tracerProviderBuilder;
+        }
+
+        /// <summary>
+        /// Adds instrumentation to the provider.
+        /// </summary>
+        /// <remarks>
+        /// Note: The type specified by <typeparamref name="T"/> will be
+        /// registered as a singleton service into application services.
+        /// </remarks>
+        /// <typeparam name="T">Instrumentation type.</typeparam>
+        /// <param name="tracerProviderBuilder"><see cref="TracerProviderBuilder"/>.</param>
+        /// <returns>The supplied <see cref="TracerProviderBuilder"/> for chaining.</returns>
+        public static TracerProviderBuilder AddInstrumentation<T>(this TracerProviderBuilder tracerProviderBuilder)
+            where T : class
+        {
+            if (tracerProviderBuilder is TracerProviderBuilderBase tracerProviderBuilderBase)
+            {
+                tracerProviderBuilderBase.AddInstrumentation<T>();
+            }
+
+            return tracerProviderBuilder;
+        }
+
+        /// <summary>
+        /// Register a callback action to configure the <see
+        /// cref="IServiceCollection"/> where tracing services are configured.
+        /// </summary>
+        /// <remarks>
+        /// Note: Tracing services are only available during the application
+        /// configuration phase.
+        /// </remarks>
+        /// <param name="tracerProviderBuilder"><see cref="TracerProviderBuilder"/>.</param>
+        /// <param name="configure">Configuration callback.</param>
+        /// <returns>The supplied <see cref="TracerProviderBuilder"/> for chaining.</returns>
+        public static TracerProviderBuilder ConfigureServices(
+            this TracerProviderBuilder tracerProviderBuilder,
+            Action<IServiceCollection> configure)
+        {
+            if (tracerProviderBuilder is TracerProviderBuilderBase tracerProviderBuilderBase)
+            {
+                tracerProviderBuilderBase.ConfigureServices(configure);
+            }
+
+            return tracerProviderBuilder;
+        }
+
+        /// <summary>
+        /// Gets the application <see cref="IServiceCollection"/> attached to
+        /// the <see cref="TracerProviderBuilder"/>.
+        /// </summary>
+        /// <param name="tracerProviderBuilder"><see cref="TracerProviderBuilder"/>.</param>
+        /// <returns><see cref="IServiceCollection"/> or <see langword="null"/>
+        /// if services are unavailable.</returns>
+        [Obsolete("Call ConfigureServices instead this method will be removed in a future version.")]
+        public static IServiceCollection? GetServices(this TracerProviderBuilder tracerProviderBuilder)
```

Comment:
I'm beginning to question if we should be less conservative here. That is, just dump `GetServices`. Likewise dump `Configure`.

Yes, it'll break folks who are currently using `OpenTelemetry.Extensions.Hosting`, but my sense of these methods are that they're less frequently used.

Reply:
@alanwest I pushed an update which removed them from SDK but I left them in Hosting and added `[Obsolete]` there. I don't feel passionately that we shouldn't remove them outright, but I feel like the Hosting API has existed for so long it would be nice for us to not break anyone. LMK what you think about that.